test/e2e/network/ingress.go

```diff
@@ -935,3 +939,220 @@ func detectNegAnnotation(f *framework.Framework, jig *e2eingress.TestJig, gceCon
 		framework.ExpectNoError(err)
 	}
 }
+
+var _ = SIGDescribe("Ingress API", func() {
+	f := framework.NewDefaultFramework("ingress")
+	/*
+		Release: v1.19
+		Testname: Ingress API
+		Description:
+		The networking.k8s.io API group MUST exist in the /apis discovery document.
+		The networking.k8s.io/v1beta1 API group/version MUST exist in the /apis/networking.k8s.io discovery document.
+		The ingresses resources MUST exist in the /apis/networking.k8s.io/v1beta1 discovery document.
+		The ingresses resource must support create, get, list, watch, update, patch, delete, and deletecollection.
+		The ingresses/status resource must support update and patch
+
+	*/
+
+	ginkgo.It("should support creating Ingress API operations", func() {
+		// Setup
+		ns := f.Namespace.Name
+		ingVersion := "v1beta1"
+		ingClient := f.ClientSet.NetworkingV1beta1().Ingresses(ns)
+
+		implementationPathType := networkingv1beta1.PathTypeImplementationSpecific
+
+		ingTemplate := &networkingv1beta1.Ingress{
+			ObjectMeta: metav1.ObjectMeta{GenerateName: "e2e-example-ing",
+				Labels: map[string]string{
+					"special-label": f.UniqueName,
+				}},
+			Spec: networkingv1beta1.IngressSpec{
+				Backend: &networkingv1beta1.IngressBackend{
+					ServiceName: "default-backend",
+					ServicePort: intstr.FromInt(8080),
+				},
+				Rules: []networkingv1beta1.IngressRule{
+					{
+						Host: "foo.bar.com",
+						IngressRuleValue: networkingv1beta1.IngressRuleValue{
+							HTTP: &networkingv1beta1.HTTPIngressRuleValue{
+								Paths: []networkingv1beta1.HTTPIngressPath{{
+									Path:     "/([a-z0-9]*)",
```

Comment:
I don't have a strong opinion either way, but would like rob's input - if we intend to promote this to a conformance test, would a pathType of prefix and a path of `/` be better?

Reply:
I think it's good to have a test like this that ensures that a regex is still valid here. If we were testing conformance of ingress controller implementations here, a `/` would certainly be easier to validate. Since that work is [happening elsewhere](https://github.com/kubernetes-sigs/ingress-controller-conformance), I'm not sure it matters. Though I'm also not familiar with what the best practices are for the kind of conformance testing we're trying to achieve here.

Reply:
if we're just testing CRUD here, simpler/better-specified seems better to me, but like I said, I don't feel that strongly

Reply:
let's change this to `/`